### ISSUE TITLE
Ability for a ServiceProvider to fail with an exception

### ISSUE THREAD

**Issue by ljacomet** [Terracotta-OSS/terracotta-apis#188]
When working on ehcache/ehcache3#1710 we simplified the lifecycle between client and server by moving bits of configuration into the entity creation. However, we had to do some configuration tricks between the entity and the corresponding state service as we cannot fail service retrieval, there is not supported exception path.

Could we consider adding a `ServiceCreationException` or similar when a `ServiceProvider` knows that the configuration cannot be honoured?

**Comment by jd0-sag** [comment]
Just to be clear, is this in `initialize` (operating on `ServiceProviderConfiguration`) or `getService` (operating on `ServiceConfiguration`)?

I assume you mean the `getService` case.

The only concern I have with this whether or not we will need a way to handle multiple `ServiceProvider` instances proving the same service type.  If so, we would probably change the rule from "only one provider per type" to "only one non-null response from `getService` per type".  In that situation, there would be 2 meanings of failure from this call.

**Comment by ljacomet** [comment]
@jd0-sag Indeed, it is `getService`. For the multi `ServiceProvider` case, we could rely (not ideal I admit) on a specific exception to mean _I handle this type but can't answer this request_ as opposed to _I am supposed to handle this but failed to for non recoverable reasons_.
